eyeshade/controllers/accounts.js

```diff
@@ -53,27 +53,79 @@ v1.getTransactions =
   handler: (runtime) => {
     return async (request, h) => {
       const account = request.params.account
-      const query1 = `select
+      let { type } = request.query
+      const args = [account]
+      let typeExtension = ''
+      type = _.isString(type) ? [type] : (type || [])
+      if (type.length) {
+        args.push(type)
+        typeExtension = 'AND transaction_type = ANY($2::text[])'
+      }
+      const query1 = `SELECT
   created_at,
   description,
   channel,
   amount,
+  from_account,
+  to_account,
+  to_account_type,
   settlement_currency,
   settlement_amount,
-  settlement_destination_type,
-  settlement_destination,
   transaction_type
-from account_transactions
-where account_id = $1
+FROM transactions
+WHERE (
+  from_account = $1
+  OR to_account = $1
+) ${typeExtension}
 ORDER BY created_at
 `
 
-      const result = await runtime.postgres.query(query1, [account], true)
-      const transactions = result.rows
+      const {
+        rows: transactions
+      } = await runtime.postgres.query(query1, args, true)
 
-      return _.map(transactions, (tx) => {
-        const omitted = _.omit(tx, (value) => value == null)
-        return Object.assign({ channel: '' }, omitted)
+      const settlementTypes = {
+        contribution_settlement: true,
+        referral_settlement: true
+      }
+      return _.map(transactions, ({
+        channel = '',
+        created_at: createdAt,
+        description,
+        from_account: fromAccount,
+        to_account: toAccount,
+        to_account_type: toAccountType,
+        amount: _amount,
+        settlement_currency: settlementCurrency,
+        settlement_amount: settlementAmount,
+        transaction_type: transactionType
+      }) => {
+        let amount = new BigNumber(_amount)
+        if (fromAccount === account) {
+          amount = amount.negated()
+        }
+        const transaction = {
+          channel: channel || '',
+          created_at: createdAt,
+          description,
+          amount: amount.toFixed(18)
+        }
+        if (settlementCurrency) {
+          transaction.settlement_currency = settlementCurrency
+        }
+        if (settlementAmount) {
+          transaction.settlement_amount = settlementAmount
+        }
+        if (settlementTypes[transactionType]) {
+          if (toAccountType) {
+            transaction.settlement_destination_type = toAccountType
+          }
+          if (toAccount) {
+            transaction.settlement_destination = toAccount
```

Comment:
basically i am just replicating exactly what was being done with the view, but in a map. so in this case only for transactions that are a type of settlement and since there is no settlement coming from an individual other than ourselves, i didn't include that condition. basically the view uses the to_account field: 
https://github.com/brave-intl/bat-ledger/blob/master/eyeshade/migrations/0001_transactions/up.sql#L47
https://github.com/brave-intl/bat-ledger/blob/master/eyeshade/migrations/0001_transactions/up.sql#L61